sklearn/pipeline.py

```diff
@@ -378,9 +392,66 @@ def _check_method_params(self, method, props, **kwargs):
                 fit_params_steps[step]["fit_predict"][param] = pval
             return fit_params_steps
 
+    def _get_step_params(self, *, step_idx, step_params, all_params):
```

Comment:
I know naming is not easy. If it "transformes" the metadata, why not naming it in that direction: _transform_metadata_for_step or similar.

Reply:
`_get_metadate_for_step` then?